client/packages/common/src/ui/components/inputs/DateTimePickers/DateTimePickerInput/DateTimePickerInput.tsx

```diff
@@ -0,0 +1,186 @@
+  onChange,
+  onError,
+  setIsOpen,
+  width,
+  label,
+  minDate,
+  maxDate,
+  showTime,
+  actions,
+  dateAsEndOfDay,
+  disableFuture,
+  displayAs,
+  error,
+  required,
+  ...props
+}: Omit<DateTimePickerProps<true>, 'onChange'> & {
+  error?: string | undefined;
+  width?: number | string;
+  label?: string;
+  onChange: (value: Date | null) => void;
+  onError?: (validationError: string, date?: Date | null) => void;
+  // This allows a calling component to know whether the date was changed via
+  // keyboard input or the picker UI
+  setIsOpen?: (open: boolean) => void;
+  showTime?: boolean;
+  actions?: PickersActionBarAction[];
+  dateAsEndOfDay?: boolean;
+  disableFuture?: boolean;
+  displayAs?: 'date' | 'dateTime';
+  required?: boolean;
+}) => {
+  const theme = useAppTheme();
+  const [internalError, setInternalError] = useState<string | null>(null);
+  const [value, setValue] = useBufferState<Date | null>(props.value ?? null);
+  const [isInitialEntry, setIsInitialEntry] = useState(true);
+  const t = useTranslation();
+  const format =
+    props.format === undefined ? (showTime ? 'P p' : 'P') : props.format;
+
+  const isDesktop = useMediaQuery('(pointer: fine)');
+
+  const updateDate = (date: Date | null) => {
+    setValue(date);
+    onChange(date);
+  };
+
+  // Max/Min should be restricted by the UI, but it's not restricting TIME input
+  // (only Date component). So this function will enforce the max/min after
+  // input
+  const handleDateInput = (date: Date | null) => {
+    if (minDate && date && date < minDate) {
+      updateDate(minDate);
+      return;
+    }
+    if (maxDate && date && date > maxDate) {
+      updateDate(maxDate);
+      return;
+    }
+
+    const dateToSave = date && dateAsEndOfDay ? DateUtils.endOfDay(date) : date;
+    updateDate(dateToSave);
+  };
+
+  return (
+    <Box display="flex">
+      <DateTimePicker
+        format={format}
+        onChange={(date, context) => {
+          const { validationError } = context;
+
+          if (validationError) {
+            const translatedError = getFormattedDateError(t, validationError);
+            if (onError) onError(translatedError, date);
+            else setInternalError(validationError ? translatedError : null);
+
+            // If there is a validation error, set internal value (so user can
+            // keep typing) but not the external one via handleDateInput
+            setValue(date);
+            return;
+          }
+          if (!validationError) {
+            setIsInitialEntry(false);
+            setInternalError(null);
+          }
+
+          handleDateInput(date);
+        }}
+        label={label}
+        slotProps={{
+          mobilePaper: { sx: getPaperSx(theme) },
+          desktopPaper: { sx: getPaperSx(theme) },
+          actionBar: {
+            actions: actions ?? ['clear', 'accept'],
+            sx: getActionBarSx(theme),
+          },
+          textField: {
```

Comment:
Huh, GH diff is showing this as new file, but actually not too many changes here - main thing was removing our text field for the display slot, and instead just styling base input to match our other inputs. This means the day/month/year sections behave more reliably, and we have accessible DOM structure.

Also pulled styles into own file (to share with time picker)

Note that with v8, you can now type in the date field on mobile as well, so calendar icon shows on both browser and tablet. There is an option to prevent typing for everyone, only open date selector, but I don't think we should move to this without validating which way works better for our users